libc/include/CMakeLists.txt

```diff
@@ -18,24 +18,14 @@ add_header(
 )
 
 macro(add_header_macro TARGET_NAME YAML_FILE DEF_FILE GEN_HDR DEPENDS)
```

Comment:
nit: Add a TODO to clean this up

Reply:
ae75b6b1c9b90bdc1335cfdd96b38accc47f248e

I tried replacing invocations of `add_header_macro` with just `add_gen_header`, but this seemed to run into issues. Not sure if every function invocation of `add_gen_header` would need explicit `DEF_FILE` and `GEN_HDR` spelled out?

Reply:
I believe they will need explicit DEF_FILE and GEN_HDR. The macro for `add_gen_header2` (lines22-29) should be used, but can be renamed to `add_gen_header`